client/gutenberg/editor/utils/media-updater/index.js

```diff
@@ -0,0 +1,148 @@
+	 */
+	updateMediaBlocks = debounce( blocks => {
+		blocks.forEach( block => {
+			if ( this.mediaBocks[ block.name ] ) {
+				this.mediaBocks[ block.name ]( block );
+			}
+
+			if ( block.innerBlocks.length ) {
+				this.updateMediaBlocks( block.innerBlocks );
+			}
+		} );
+	} );
+
+	/**
+	 * Preload an image given its URL so it can be rendered without flickering
+	 * @param {String} imageUrl URL of the image to preload
+	 * @returns {Promise} Promises that resolves when the image is preloaded
+	 */
+	preloadImage = imageUrl =>
+		new Promise( ( resolve, reject ) => {
+			const preloadImage = new Image();
+			preloadImage.src = imageUrl;
+			preloadImage.onload = resolve;
+			preloadImage.onerror = reject;
+		} );
+
+	/**
+	 * Update the block attribute storing a single image URL
+	 * @param {Object} block Block to update
+	 * @param {Object} attrNames Mapping defining the name of the attributes to update
+	 */
+	updateSingeImageBlock = async ( block, attrNames = {} ) => {
+		attrNames = {
+			id: 'id',
+			url: 'url',
+			...attrNames,
+		};
+
+		const { id, url } = attrNames;
+		const { siteId } = this;
+		const { clientId, attributes } = block;
+
+		const media = MediaStore.get( siteId, attributes[ id ] );
+
+		if ( media ) {
+			// If image was deleted in the Media Modal, we delete it in the block.
+			if ( media.status === 'deleted' || ! media.URL ) {
+				updateBlockAttributes( clientId, {
+					[ id ]: undefined,
+					[ url ]: undefined,
+				} );
+			} else {
+				// To avoid an undesirable flicker when the image hasn't yet been loaded,
+				// we preload the image before rendering.
+				await this.preloadImage( media.URL );
+
+				updateBlockAttributes( clientId, { [ url ]: media.URL } );
+			}
+		}
+	};
+
+	/**
+	 * Update the block attribute storing a list of images
+	 * @param {Object} block Block to update
+	 * @param {Object} attrNames Mapping defining the name of the attributes to update
+	 */
+	updateMultipleImagesBlock = async ( block, attrNames = {} ) => {
+		attrNames = {
+			images: 'images',
+			id: 'id',
+			url: 'url',
+			...attrNames,
+		};
+
+		const { images, id, url } = attrNames;
+		const { siteId } = this;
+		const { clientId, attributes } = block;
+
+		const currentImages = attributes[ images ];
+		const updatedImages = [];
+
+		for ( const image of currentImages ) {
+			const media = MediaStore.get( siteId, image[ id ] );
+
+			// Deleted images are not included in the new images attribute so
+			// they are removed from the block.
+			if ( media && media.status !== 'deleted' && media.URL ) {
+				// To avoid an undesirable flicker when the image hasn't yet been loaded,
+				// we preload the image before rendering.
+				await this.preloadImage( media.URL );
+
+				updatedImages.push( {
+					...image,
+					[ url ]: media.URL,
+				} );
+			}
+		}
+
+		updateBlockAttributes( clientId, { [ images ]: updatedImages } );
+	};
+
+	// Block name -> Function that updates the media
+	mediaBocks = {
```

Comment:
An explicit mapping works in the short term 👍

We may want to chat with @Automattic/luna and core folks, maybe @gziolo @aduth, if there's a way to detect blocks that use images, and maybe provide a decent fallback option

Reply:
It looks like in core we don't refresh content at all when any change in the media library happens. I tried the following:
- delete an image using media library, images are missing as soon as you refresh the page
- edit an image using media library - only the image in the currently selected block gets updated, however the old image still works properly for other blocks

So, the short answer is we don't have support for it at the moment at all :(

Regarding finding the ways, I don't think it is straightforward because urls are stored as strings in `attributes` section and we simply inline components which handle them. I know that there were some work to strip out image ids when exporting/importing reusable blocks, so maybe @mcsf or @youknowriad would know more how to automate it.

Reply:
Thanks @gziolo for the extra context. I suppose we'll keep thinking about this for core's use case as well. 🤔Maybe we could build something that could query for block functionality. Say, of an editors registered blocks, which ones involve the media library, and it'd return this list or something similar. Or may return a list of blocks instances that satisfy the feature query. I'm guessing we may have other use cases like that.